Add E2E tests to cover Publish dropdown options

octaedro opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR adds tests to cover the Publish dropdown options. It will test the `Schedule` a product publication, the `Copy to new draft` option and the `Move to trash` functionallity.   

![Screenshot 2024-04-17 at 1 07 56 PM](https://github.com/woocommerce/woocommerce/assets/1314156/4b488a32-1c37-4997-8829-f843223d895a)

Closes #46593.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch
2. Run `cd plugins/woocommerce && pnpm env:start`
3. Then run `USE_WP_ENV=1 pnpm playwright test --config=tests/e2e-pw/playwright.config.js` if you add the relative path behind it will run the file you want. I would also add `--headed` to see visually what's going on. In this case, the command should look like `USE_WP_ENV=1 pnpm playwright test --config=tests/e2e-pw/playwright.config.js ./tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js --headed`
4. All the tests should pass.

More details [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/README.md).

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

octaedro committed:
Add e2e tests for publish options

octaedro committed:
Add changelog

louwie17 commented on plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js:
```diff
@@ -80,3 +80,70 @@ test( 'can update the general information of a product', async ( {
 	} );
 } );
+
+test( 'can schedule a product publication', async ( { page, product } ) => {
```
This test schedules a product, duplicates a product and removes a product.
Should we not move these into individual tests?

octaedro replied:
Good idea! Done in https://github.com/woocommerce/woocommerce/pull/46658/commits/03ac99954746babfc0b1459b4bd7c0d1c61694e4

louwie17 reviewed (changes requested):
Thanks for adding this Fernando, the E2E test ran fine locally, I just think we should split it up in a couple different tests if possible?

Also in relation to your testing instructions:

> Run our E2E tests. All the tests should pass.

It be great if you could be more specific here in the future, for example: list the commands that are required for E2E tests. Not everyone on the team will be aware of how to do this and it would save the team a lot of time if they don't need to go looking for it. Linking to the E2E docs would also help.

octaedro committed:
Create different tests for options

octaedro commented:
Thank you @louwie17 for your review.

I updated the testing instructions and addressed the suggestion you made.
Could you take another look at this PR?

louwie17 commented on plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js:
```diff
@@ -80,3 +80,84 @@ test( 'can update the general information of a product', async ( {
+			page.getByLabel( 'Dismiss this notice' ).first()
+		).toContainText( 'Product scheduled for' );
+	} );
+	test( 'can dupicate a product', async ( { page, product } ) => {
```
```suggestion
	test( 'can duplicate a product', async ( { page, product } ) => {
```
Small spelling mistake

louwie17 reviewed (approved):
LGTM 🚀 Thanks for the updates @octaedro, just a small spelling mistake left but otherwise all is good.

octaedro committed:
Update plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js

Co-authored-by: louwie17 <lourensschep@gmail.com>
